src/Debugger.php

```diff
@@ -72,39 +104,58 @@ public function startup(object $event): void
         }
     }
 
+    /**
+     * Shuts down debugger and collectors. Collected data is flushed to storage.
+     */
     public function shutdown(): void
     {
         if (!$this->isActive()) {
             return;
         }
 
         try {
-            $collectedData = array_map(
-                static fn (CollectorInterface $collector) => $collector->getCollected(),
-                $this->collectors
-            );
-
-            /** @var array[] $data */
-            [$data, $objectsMap] = $this->dataNormalizer->prepareDataAndObjectsMap($collectedData, 30);
-
-            /** @var array $summary */
-            $summary = $this->dataNormalizer->prepareData($this->collectSummaryData(), 30);
-
-            $this->storage->write($this->getId(), $data, $objectsMap, $summary);
+            $this->flush();
         } finally {
-            foreach ($this->collectors as $collector) {
-                $collector->shutdown();
-            }
-            $this->id = null;
+            $this->stop();
         }
     }
 
-    public function stop(): void
+    /**
+     * Stop debugger and collectors. Collected data isn't flushed to storage.
+     */
+    public function terminate(): void
```

Comment:
I'd suggest `deactivate` or `disable`. Terminate means that object would be accessible any longer

Reply:
The difference with `shutdown` is only about flushing data. Maybe `pause` is better here?

Reply:
By the way, should it be as public api?